[Fleet][Endpoint][RBAC V2] Update fleet router and config to allow API access via RBAC controls

ashokaditya opened the pull request:
## Summary

> [!Note]
> This PR is adding changes only to some of `api/fleet/package_policies`
API routes, there will be subsequent PRs after this to update
`api/fleet/epm/packages`, `api/fleet/agent_policeis` and,
`api/fleet/agent_status`.

This PR introduces the framework needed in fleet in order to be able to
support Package level Privileges - meaning: if a user does not have
authorization granted via Fleet and/or Integration privileges, then
package level privileges are check and API access granted. When access
is granted based on Package Privileges, the data is also validated to
ensure that it is limited to the integration package names that were
given authorization to the API.

The following APIs were updated to leverage this new framework:

- Integration Package Policy list API
- Integration Package Policy get one API
- Integration Package Policy update one API
- Integration Package Policy bulk get API

> ℹ️ these API were updated in support of Endpoint use cases needed for
v8.7.

Example of API error for Package policies api:

```json5
{
    "statusCode": 403,
    "error": "Forbidden",
    "message": "Authorization denied to [package.name=fleet_server]. Allowed package.name's: endpoint"
}
```
___________

To test:
1. Log in as `elastic`/superuser and create some agent policies.
1. Under `Stack Management`, create a role `policy_role` with the
following RBAC settings. **DO NOT** select `Fleet -> All` or toggle
`Integrations`. Leave those RBAC toggles set to `None`
<img width="610" alt="Screenshot 2022-11-16 at 14 45 15"
src="https://user-images.githubusercontent.com/1849116/202196962-9123e380-3b8f-4d52-97f9-8af895fb4c26.png">

2. Create a user e.g. `policy_user` and assign them _only_ the above
role. **NOT** `superuser`.

3. Login with this user and navigate to
`app/security/administration/policy` or curl/postman.
4. Expect to see the following:
- GET `api/fleet/epm/packages?category=security` should return a `403`
status.
- GET
`api/fleet/package_policies?page=1&perPage=10&kuery=ingest-package-policies.package.name%3A%20endpoint`
should return a list of policies.
- GET `/api/fleet/package_policies/<packagePolicyId>` should return a
`200` and a signle item that has the policie's details. Note that the
package name of this item is `endpoint`.
- there should be a POST API request matching
`api/fleet/agent_policies/_bulk_get`, and should return a `403`.
5. With `Policy Management` RBAC set to `All`
- PUT
`http://localhost:5601/api/fleet/package_policies/<packagePolicyId>`
should return a `200` with the updated policy details as response

### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

- [x] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

Co-authored-by: Paul Tavares <paul.tavares@elastic.co>

ashokaditya committed:
update fleet router config to allow route access independent of tags

fixes elastic/security-team/issues/4926
ref elastic/security-team/issues/5539

ashokaditya committed:
update fleet router config to allow route access independent of tags

fixes elastic/security-team/issues/4926
ref elastic/security-team/issues/5539

ashokaditya committed:
update fleet router config to allow route access independent of tags

fixes elastic/security-team/issues/4926
ref elastic/security-team/issues/5539

paul-tavares committed:
Merge remote-tracking branch 'ashokaditya/task/olm-policy-list-api-rbac-4926' into pr/ash/task/olm-policy-list-api-rbac-4926

# Conflicts:
#	x-pack/plugins/fleet/server/routes/agent_policy/index.ts
#	x-pack/plugins/fleet/server/routes/epm/index.ts
#	x-pack/plugins/fleet/server/routes/package_policy/index.ts
#	x-pack/plugins/fleet/server/routes/security.ts
#	x-pack/plugins/fleet/server/services/package_policy.ts

paul-tavares commented on x-pack/plugins/fleet/server/routes/security.ts:
```diff
@@ -110,6 +112,116 @@ export async function getAuthzFromRequest(req: KibanaRequest): Promise<FleetAuth
 }
 
+// needed from Read routes
+export const readEndpointPackagePrivileges: DeepPartialTruthy<FleetAuthz> = Object.freeze({
```
Suggestion: name the constant in upper case: `READ_ENDPINT_PACKAGE_PRIVILEGES` (we normally name constants in upper case using snake_case).
Same for the other const below.

paul-tavares commented on x-pack/plugins/fleet/server/routes/security.ts:
```diff
@@ -110,6 +112,116 @@ export async function getAuthzFromRequest(req: KibanaRequest): Promise<FleetAuth
+      return acc;
+    }, []);
+
+  if (requiredAuthz.any && !requiredAuthz.all) {
+    return getBoolListFromPaths(requiredAuthz.any).some((v) => v);
+  } else if (requiredAuthz.all) {
+    return getBoolListFromPaths(requiredAuthz.all).every((v) => v);
+  } else if (requiredAuthz.any && requiredAuthz.all) {
+    // integration privileges should be all true or any of the endpoint privileges should be true
+    // e.g. epm/BULK_INSTALL_PATTERN
+    return (
+      getBoolListFromPaths(requiredAuthz.any).some((v) => v) ||
+      getBoolListFromPaths(requiredAuthz.all).every((v) => v)
+    );
+  }
+  return false;
```
The `all` and `any`, when defined together, should act as a `AND`, not an `OR`.  Here is a suggestion on simplifying this code bit here:

```typescript

// First validate that they have `all` - if not, exit with `false`
if (requiredAuthz.all && requiredAuthz.all.length && !getBoolListFromPaths(requiredAuthz.all).every((v) => v)) {
  return false;
}

// Now validate that they (also) have `any` - if not, then exit with `false`
if (requiredAuthz.any && requiredAuthz.any.length && !getBoolListFromPaths(requiredAuthz.any).some((v) => v)) {
  return false;
}

// If we reached this spot, then user has all the authz necessary - return `true`
return true;
```

ashokaditya replied:
Yeah, that is deliberate with the `OR` here when both `all` and `any` are defined, to allow access to `POST` on `api/fleet/epm/packages/_bulk` from the Security side as well as Fleet side based on if only Fleet privileges were on or Security sub-privileges were on. 

That being said, I'm not entirely sure if we want to allow installing and upgrading packages from the Security side and fleet side independently based on RBAC. 

See [here in the epm route config](https://github.com/elastic/kibana/pull/145361/files/6c6eb1a7bb51203a530ecf1a250006d17e0733ce#diff-2d62e711521fc58fe69b5113f50cdda725b3fc21d52e6a2aac46245fabc08b97R156-R165)

ashokaditya replied:
Basically, when multiple _fleet_ privileges are required they are grouped in `all` as _all_ of them are required to access the route, and the security sub-privilegs that should work independently of fleet are always grouped in `any`. So even if the fleet privileges are not set, we can still access the route when any of the security-sub privileges are true. For instance, with the API in question, the router configuration is:

```json5
router.post(
    {
      path: EPM_API_ROUTES.BULK_INSTALL_PATTERN,
      validate: BulkUpgradePackagesFromRegistryRequestSchema,
      fleetAuthz: {
            integrations: { installPackages: true, upgradePackages: true },
          },
    },
    bulkInstallPackagesFromRegistryHandler
  );
```

changes to the following with additional security sub-privileges:

```json5
router.post(
    {
      path: EPM_API_ROUTES.BULK_INSTALL_PATTERN,
      validate: BulkUpgradePackagesFromRegistryRequestSchema,
      fleetAuthz: (fleetAuthz: FleetAuthz): boolean =>
        validateSecurityRbac(fleetAuthz, {
          all: {
            integrations: { installPackages: true, upgradePackages: true },
          },
          any: {
            ...writeEndpointPackagePrivileges,
          },
        }),
    },
    bulkInstallPackagesFromRegistryHandler
  );
```

So when fleet privileges are toggled we need both `installPackages` and `upgradePackages` to be `true`.

paul-tavares replied:
Ok. I see. since we don't currently have the need for `AND`, I'm ok leaving as you have it.   The `AND` allows for use cases where a minimum set of privileges is required ++ at least one other privileges from a list (`any`). 


________

🤔 

I'm wondering, then, if the `all` / `any` make sense then. So what we are expressing here with `validateSecurityRbac ` is that:

1. Fleet privileges required to access API
    -- OR : ANY one of --
 2. Package Privileges required (and in this case, data will be limited to that package)

Maybe the second argument to `validateSecurityRbac()` should be:

```typescript
{
  // The fleet set of privileges required for this API. All MUST be `true` in order to be granted authz
  fleet: {},

  // If no `fleet` authz (above), then ANY one of the following package privileges is set to `true`,
  // then allow access to API, but limit data to the set of packages (name's) that were defined here.
  packages: {}
}
```

paul-tavares reviewed (commented):
Did a review and left some comments. 

I also checked the PR out and although the APIs are returning `200` (success), the Package Policy list API does not actually return any data, although endpoint policies existed in my testing. My guess is because its using the scoped SO client instead of the internal one

ashokaditya committed:
rename

ashokaditya committed:
simplify logic

juliaElastic commented:
LGTM so far, the missing piece is making sure that the `endpoint` roles only allow modifying endpoint package policies/agent policies.

ashokaditya committed:
cleanup

ashokaditya committed:
fix tests

juliaElastic commented on x-pack/plugins/fleet/server/routes/epm/handlers.ts:
```diff
@@ -66,7 +66,7 @@ const CACHE_CONTROL_10_MINUTES_HEADER: HttpResponseOptions['headers'] = {
   undefined,
   TypeOf<typeof GetCategoriesRequestSchema.query>
-> = async (context, request, response) => {
+> = async (_context, request, response) => {
```
can this be reverted?

ashokaditya replied:
Sure I'll change it back if you like. It's not being used though so I renamed it for clarity.

ashokaditya committed:
Merge branch 'main' into task/olm-policy-list-api-rbac-4926

ashokaditya committed:
Merge branch 'main' into task/olm-policy-list-api-rbac-4926

paul-tavares committed:
Merge remote-tracking branch 'ashokaditya/task/olm-policy-list-api-rbac-4926' into pr/ash/task/olm-policy-list-api-rbac-4926

# Conflicts:
#	x-pack/plugins/fleet/server/routes/agent_policy/index.ts
#	x-pack/plugins/fleet/server/routes/epm/index.ts
#	x-pack/plugins/fleet/server/routes/package_policy/index.ts
#	x-pack/plugins/fleet/server/routes/security.test.ts
#	x-pack/plugins/fleet/server/routes/security.ts
#	x-pack/plugins/fleet/server/services/package_policy.ts

ashokaditya committed:
Merge branch 'main' into task/olm-policy-list-api-rbac-4926

ashokaditya committed:
Merge branch 'main' into task/olm-policy-list-api-rbac-4926

ashokaditya committed:
Merge branch 'main' into task/olm-policy-list-api-rbac-4926

paul-tavares committed:
Retrieve correct SO client when Authz are NOT granted via Fleet privileges

paul-tavares committed:
Merge remote-tracking branch 'ashokaditya/task/olm-policy-list-api-rbac-4926' into pr/ash/task/olm-policy-list-api-rbac-4926

ashokaditya committed:
update types and use `calculateRouteAuthz`

for policy service

ashokaditya committed:
update authzwrapper to use `calculateRouteAuthz`

paul-tavares committed:
Update logic in `calculateRouteAuthz()` so that each package privileges is checked individually

paul-tavares committed:
Revert changes to agent policy and epm APIs (keeping scope small)

paul-tavares committed:
backout change to epm categories route handler

paul-tavares committed:
Merge remote-tracking branch 'upstream/main' into pr/ash/task/olm-policy-list-api-rbac-4926

paul-tavares committed:
Refactor package policy routes and Plugin route context to use `calculateRouteAuthz`

paul-tavares committed:
Adjusted package policy service to use `calculateRouteAuthz`

paul-tavares committed:
Remove unused code from `security.ts`

paul-tavares committed:
Refactor: re-organized server security module

paul-tavares committed:
Move fleet router tests to its own file

paul-tavares committed:
Fix mocks and security tests

paul-tavares committed:
add to tests for `calculateRouteAuthz()`

paul-tavares committed:
Update package policy routes to get SO object from `fleet.getSoClient()`

ashokaditya committed:
unify logic for gating access

ashokaditya committed:
fix type check

ashokaditya committed:
fix route path pattern match

refs 60f35558bb1ef1fb98ae360ee656bd0a511ec604

ashokaditya committed:
Merge branch 'main' into task/olm-policy-list-api-rbac-4926

ashokaditya committed:
fix mock path after refactor

refs 60f35558bb1ef1fb98ae360ee656bd0a511ec604

ashokaditya committed:
use main route pattern

refs 946d5fdfba80d0d0227c6eac0d501df79d52ef82

ashokaditya committed:
Merge branch 'main' into task/olm-policy-list-api-rbac-4926

ashokaditya committed:
parse routePath instead

refs 946d5fdfba80d0d0227c6eac0d501df79d52ef82

ashokaditya committed:
parse routePath instead

refs 946d5fdfba80d0d0227c6eac0d501df79d52ef82

ashokaditya committed:
Merge branch 'main' into task/olm-policy-list-api-rbac-4926

ashokaditya committed:
limit access to requested packages

ashokaditya committed:
limit access to data based on allowed packages

ashokaditya committed:
package specific list results - part 2

paul-tavares committed:
Merge remote-tracking branch 'ashokaditya/task/olm-policy-list-api-rbac-4926' into pr/ash/task/olm-policy-list-api-rbac-4926

# Conflicts:
#	x-pack/plugins/fleet/server/services/security/route_required_authz.ts

paul-tavares commented on x-pack/plugins/fleet/server/routes/package_policy/handlers.ts:
```diff
@@ -43,16 +43,68 @@ import { simplifiedPackagePolicytoNewPackagePolicy } from '../../../common/servi
+        });
+      }
+
+      // list allowed package policy items or policies with no package name
+      return response.ok({
+        body: {
+          items: items.filter((item) => {
+            const pkgName = item?.package?.name;
+            // include policies with allowed package names
+            const allowedPackages = Object.entries(packagePolicyNames).reduce<string[]>(
+              (acc, curr) => {
+                if (curr[1]) acc.push(curr[0]);
+                return acc;
+              },
+              []
+            );
+            if (pkgName) {
+              return allowedPackages.includes(pkgName);
+            }
+            // or policies with no package name
+            return item;
+          }),
+          total,
+          page,
+          perPage,
+        },
+      });
```
I don't understand what this code is attempting to do?

ashokaditya replied:
I wasn't sure about this actually, but here it is filtering put policy items based on allowed package names. In case there are policies without a package name (not sure if there ever is such a case but at leat the type says it's optional) it allows it in the list.

paul-tavares replied:
What you mean "put"? this is the `get list` api.

I'm going to remove this. Also - for now, although the Type definition indicates that `package` is optional, I don't know of a use case where it might be undefined - perhaps when Agent is used in standalone mode?

For now, we I propose we go with: if `package` is not defined, then we will error out. This seems like the right thing to do being that we are adding access based on `package privileges`, so a package is a requirement. 

Note: I'm also changing a bit your logic around validation and error out. We should exit the moment we encounter an invalid `package.name` and not attempt to loop through all of the data. That because the API's in fleet do not bound the `perPage` value, so an API request could theoretically set this to a very large value and I don't think we should spend time collecting all of the invalid package name. New implementation will just report the first package name that is not authorized and also provide the list of package name that IS authorized.

paul-tavares committed:
Update error message for Update package policy handler

paul-tavares committed:
Re-implement package policy api handlers data validation against allowed package names

paul-tavares committed:
add `shouldHandlePostAuthRequest()` check back into fleet router middleware

paul-tavares committed:
Add check for kibana security enabled to route handler wrapper

paul-tavares committed:
Merge remote-tracking branch 'upstream/main' into pr/ash/task/olm-policy-list-api-rbac-4926

paul-tavares committed:
Add supertest logger for better messages

paul-tavares committed:
Add `supertest` middleware in bulk_upgrade tests

paul-tavares committed:
Trying again: code to capture failing test API response body

ashokaditya committed:
remove redundant `registerOnPostAuth`

ashokaditya committed:
update epm bulk_upgrade FTR tests

ashokaditya committed:
log router access denied

ashokaditya committed:
cleanup

refs 959a0f1e1b81bdf86d907aaf42439d3dd63a0f47
refs b65997276ca8ec1e3325047783dc467c48be9f3b
refs 0a34ae697036fec845a95e1d2727dc7bcd0b9ece
refs f98858c7262f30c973e2d7a6abad7b81bc199dcf

ashokaditya committed:
Merge branch 'main' into task/olm-policy-list-api-rbac-4926

elasticmachine commented:
Pinging @elastic/security-onboarding-and-lifecycle-mgt (Team:Onboarding and Lifecycle Mgt)

paul-tavares commented on x-pack/test/fleet_api_integration/apis/epm/bulk_upgrade.ts:
```diff
@@ -51,13 +51,15 @@ export default function (providerContext: FtrProviderContext) {
           .post(`/api/fleet/epm/packages/_bulk`)
           .auth(testUsers.fleet_all_int_read.username, testUsers.fleet_all_int_read.password)
           .set('kbn-xsrf', 'xxxx')
+          .send({ packages: ['multiple_versions', 'overrides'] })
```
The reason for these having to be updated to include a valid `body` is because Authz is now being done in the handler (or handler wrapper) which is after the core HTTP framework validates the request using the route's schema. Without these changes, we were getting `400` errors

nchaulet replied:
Should we not do authorization before validation?

paul-tavares replied:
Hi @nchaulet ,

Ideally, yes it should. However, the Kibana Router framework does not provide for an easy way to do that. They have middleware that we can register for post Auth/pre validation, but it does not provide you with enough information to be able do authz - specifically: we would need access to the route registration configuration, especially the information around what `path` pattern the route was registered with. 

I actually took a note to create a post to Kibana security slack channel asking them to support a `authz(): boolean | Promise<Boolean>` option on route registration. Something like:

```typescript
 router.post(
    {
      path: '/api/someplugin/{token}/view',

      authz: (): boolean => { }

    },
    async () => {}
  );
```

If have them call `authz` automatically after `auth` is done if `auths` was defined on the

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

nchaulet commented on x-pack/plugins/fleet/server/plugin.ts:
```diff
@@ -356,18 +374,32 @@ export class FleetPlugin
             return deps.spaces.spacesService.getSpaceId(request);
           },
+
+          async getSoClient() {
```
it seems weird to me that method return both a saved object client and some limited packages, should we get two property on the fleet context?

````
get soClient()
get limitedToPackages()
```

paul-tavares replied:
@ashokaditya - can you make this change tomorrow?

ashokaditya replied:
done 2886e3dc4a2f459c0e515537c5a01fd2556c96fb

nchaulet commented on x-pack/plugins/fleet/server/plugin.ts:
```diff
@@ -356,18 +374,32 @@ export class FleetPlugin
+          async getSoClient() {
+            if (routeAuthz && routeAuthz.granted) {
+              return {
+                client: routeAuthz.grantedByFleetPrivileges
```
I am wondering if there is any value of not always using the internal SO client

paul-tavares replied:
🤔 it probably would not hurt to always use the internal client. If that was to be the case, the we likely would not need this property on the Route context. I have not spent time thinking about it, but if you feel it is a safe change, we make it in this PR.  Let us know

nchaulet replied:
Yes I think it's safe change to make, and it make things simpler to use the same saved object client.

ashokaditya replied:
done 2886e3dc4a2f459c0e515537c5a01fd2556c96fb

paul-tavares replied:
@juliaElastic ,

Re: 

> Is the agent policy API going to filter out the package policies that do not belong to `endpoint` package?

I'm not sure yet about how this one should be implemented and it warrants some discussions. Our Endpoint use cases don't need `update/create/delete` access, so for us, it would be ok to filter out any package policy that does not match the authz package names. But for cases (in the future or for other integrations) where Update/Delete is needed, that gets more complicated.

@ashokaditya,
we need to explore each one of these individually and see if we can come up with good suggestions on the approach. We should use the tracking issues you created to expand upon this.

juliaElastic replied:
That's a good point, I think the problem is that the Agent policy is a resource that doesn't belong to one package only. 
We don't have to solve now for use cases that require edit access.

paul-tavares replied:
Last comment by Julia above is for this comment/response by me here:

https://github.com/elastic/kibana/pull/145361#issuecomment-1348690980

_(Sorry. Coffee had not kicked in yet, and I initially posted by response in the wrong thread and then correct it)_

ashokaditya committed:
use internalSoClient

review comments (@nchaulet, @paul-tavares)

ashokaditya committed:
Merge branch 'main' into task/olm-policy-list-api-rbac-4926

juliaElastic commented on x-pack/plugins/fleet/server/routes/package_policy/handlers.ts:
```diff
@@ -43,16 +43,74 @@ import { simplifiedPackagePolicytoNewPackagePolicy } from '../../../common/servi
     );
+
+    // specific to package-level RBAC
+    if (limitedToPackages && limitedToPackages.length) {
```
this block seems to be repeated a few times, can we extract to a function?

ashokaditya replied:
Good idea. We'll be using similar logic in other API routes (elastic/security-team/issues/5620). Where do you suggest we should place this function so it can be imported where needed?  I think we should also have a generic `validatePolicyDataIsScopedToAllowedPackageNames` besides `validatePolicyDataIsScopedToAllowedPackageNames` in the same place, that takes in the agent policy, package policy or epm data to compute this.

something like:
```json5
const validatePolicyDataIsScopedToAllowedPackageNames = (
   data:  PackagePolicy[]| AgentPolicy[] | PackageInfo ,
   allowedPackageNames: string[]): string | undefined
): string | undefined => {
// validate if data is scoped to package names
}

const validateLimitedPackages(
   limitedPackages: string[] | undefined, 
   data: PackagePolicy[] | AgentPolicy[] | PackageInfo 
): IKibanaResponse<any> | undefined => {
    const validationResult = validatePolicyDataIsScopedToAllowedPackageNames(
        data,
        limitedToPackages
      );
      if (validationResult) {
        return response.forbidden({
          body: {
            message: validationResult,
          },
        });
      }
}
```

juliaElastic replied:
you could place it under `fleet/server/routes` or the closest common ancestor

ashokaditya replied:
I cleaned up some of the duplicated logic instead 68e366aa4a4805db4337ee0371de118741eb8e3e.

juliaElastic commented:
It would be good to add API integration tests to test a few scenarios of existing and new RBAC authorization on API endpoints. I think we already have a few Cypress tests that cover the existing Fleet authorization.

juliaElastic commented:
> there should be a POST API request matching api/fleet/agent_policies/_bulk_get, and should have a list of policies as well.

Is the agent policy API going to filter out the package policies that do not belong to `endpoint` package?

ashokaditya commented:
> > there should be a POST API request matching api/fleet/agent_policies/_bulk_get, and should have a list of policies as well.
> 
> Is the agent policy API going to filter out the package policies that do not belong to `endpoint` package?

We're going to add those changes in subsequent PRs. elastic/security-team/issues/5620.

I've updated the PR description now.

ashokaditya committed:
move internalSoClient out of `epm`

Now that we're using the internal SO client in routes other than `epm` we decided to make it available directly on the fleet context.

review comments (@juliaElastic)

ashokaditya commented on x-pack/plugins/fleet/server/routes/package_policy/handlers.ts:
```diff
@@ -43,16 +43,74 @@ import { simplifiedPackagePolicytoNewPackagePolicy } from '../../../common/servi
 > = async (context, request, response) => {
-  const soClient = (await context.core).savedObjects.client;
+  const fleetContext = await context.fleet;
+  const soClient = fleetContext.epm.internalSoClient;
```
done a186363be78da4318460dcb64d05444a99a3f387

paul-tavares commented:
@juliaElastic ,

Re: 

> Is the agent policy API going to filter out the package policies that do not belong to `endpoint` package?

I'm not sure yet about how this one should be implemented and it warrants some discussions. Our Endpoint use cases don't need `update/create/delete` access, so for us, it would be ok to filter out any package policy that does not match the authz package names. But for cases (in the future or for other integrations) where Update/Delete is needed, that gets more complicated.

@ashokaditya,
we need to explore each one of these individually and see if we can come up with good suggestions on the approach. We should use the tracking issues you created to expand upon this.

ashokaditya committed:
cleanup logic for limitedToPackages

review comments (@juliaElastic)

ashokaditya committed:
Merge branch 'main' into task/olm-policy-list-api-rbac-4926

ashokaditya committed:
Merge branch 'main' into task/olm-policy-list-api-rbac-4926

paul-tavares reviewed (commented):
@ashokaditya ,
I took a look through and am good with the changes. I'm not going to approve it just because I worked with you on this. I'll defer approval to the Fleet team

ashokaditya committed:
Merge branch 'main' into task/olm-policy-list-api-rbac-4926

ashokaditya committed:
Merge branch 'main' into task/olm-policy-list-api-rbac-4926

juliaElastic commented:
The code looks good to me, I'll do a few manual tests locally before I approve.

juliaElastic commented:
I noticed something strange, I created a role that has access to Fleet:All and Integrations:Read, and I'm getting a permission denied error on Fleet/Agents screen.
The same role has access on the dev-oblt cluster, so looks like a regression from this pr.

<img width="1775" alt="image" src="https://user-images.githubusercontent.com/90178898/207604018-e10267ad-5868-4334-963d-78052bead8fa.png">

ashokaditya committed:
undo changes to `api/fleet/check-permissions`

refs eecb7401a6230d763e143d30aaed438e24cc5137

juliaElastic reviewed (approved):
LGTM, tested locally with the endpoint role and existing fleet/integrations role

ashokaditya committed:
cleanup

refs eecb7401a6230d763e143d30aaed438e24cc5137
refs e04a741b100a0117cdf6cff28885f8515a454eb6

nchaulet reviewed (approved):
LGTM 🚀

ashokaditya committed:
Merge branch 'main' into task/olm-policy-list-api-rbac-4926
